ext/nnstreamer/tensor_decoder/tensordec-directvideo.c

```diff
@@ -30,47 +30,126 @@
 #include <glib.h>
 #include <gst/video/video-format.h>
 #include <nnstreamer_plugin_api_decoder.h>
+#include <nnstreamer_plugin_api.h>
 #include <nnstreamer_log.h>
 #include "tensordecutil.h"
 
 void init_dv (void) __attribute__ ((constructor));
 void fini_dv (void) __attribute__ ((destructor));
 
 #define DECODER_DV_VIDEO_CAPS_STR \
-    GST_VIDEO_CAPS_MAKE ("{ RGB, BGRx, GRAY8 }") \
+    GST_VIDEO_CAPS_MAKE ("{ I420, GRAY8, RGB, BGR, RGBx, BGRx, xRGB, xBGR, RGBA, BGRA, ARGB, ABGR }") \
     ", views = (int) 1, interlace-mode = (string) progressive"
 
+/**
+ * @brief The supported video formats
+ */
+typedef enum
+{
+  DIRECT_VIDEO_FORMAT_UNKNOWN = 0,
+
+  /* Single Channel, Default: GRAY8 */
+  DIRECT_VIDEO_FORMAT_I420 = 1,
+  DIRECT_VIDEO_FORMAT_GRAY8 = 2,
+
+  /* 3 Channels, Default: RGB */
+  DIRECT_VIDEO_FORMAT_RGB = 3,
+  DIRECT_VIDEO_FORMAT_BGR = 4,
+
+  /* 4 Channels, Default: BGRx */
+  DIRECT_VIDEO_FORMAT_RGBx = 5,
+  DIRECT_VIDEO_FORMAT_BGRx = 6,
+  DIRECT_VIDEO_FORMAT_xRGB = 7,
+  DIRECT_VIDEO_FORMAT_xBGR = 8,
+  DIRECT_VIDEO_FORMAT_RGBA = 9,
+  DIRECT_VIDEO_FORMAT_BGRA = 10,
+  DIRECT_VIDEO_FORMAT_ARGB = 11,
+  DIRECT_VIDEO_FORMAT_ABGR = 12,
+} direct_video_formats;
+
+/**
+ * @brief Data structure for direct video options.
+ */
+typedef struct
+{
+  /* From option1 */
+  direct_video_formats format;
+} direct_video_ops;
+
+/**
+ * @brief List of the formats of direct video
+ */
+static const char *dv_formats[] = {
+  [DIRECT_VIDEO_FORMAT_UNKNOWN] = "UNKNOWN",
+  [DIRECT_VIDEO_FORMAT_I420] = "I420",
```

Comment:
Q. Does tensor support this type (I420)?

Reply:
it is impossible with rank 4 currently :) it's removed.